app/controllers/admin/users_controller.rb

```diff
@@ -28,11 +28,19 @@ def destroy
   private
     # Use callbacks to share common setup or constraints between actions.
     def set_user
-      @user = User.includes(:projects).find(params[:id])
+      @user = admin_user_policy_scope.includes(:projects).find(params[:id])
+      authorize @user
     end
 
     # Only allow a trusted parameter "white list" through.
     def user_params
       params.fetch(:user,{}).permit(:email, :name, :initials)
     end
+
+    def admin_user_policy_scope
+      # FIXME because it's a non-conventional policy class name, have to manually flag the verification
+      # is this the best way?
+      @_pundit_policy_scoped = true
+      AdminUserPolicy::Scope.new(pundit_user, User).resolve
```

Comment:
I guess I got your point, but you need to put this within `UserPolicy` like this:

```
class UserPolicy < ApplicationPolicy
  def show?
    if user.admin?
      Project.find(record.id)
    else
      context.current_project.users.find_by_id(record.id)
    end
  end

  class Scope < Scope
    def resolve
      if user.admin?
        all
      else
        context.current_project.users
      end
    end
  end
end
```